yunwoo/programmers-1/햄버거-만들기.js

```diff
@@ -0,0 +1,24 @@
+function solution(ingredient) {
+  let result = 0;
+  const temp = [];
+  const hamburger = "1231";
+
+  for (let i = 0; i < ingredient.length; i++) {
+    temp.push(ingredient[i]);
+
+    const checkLastFouringredient = temp.slice(-4).join("");
+
+    if (checkLastFouringredient === hamburger) {
+      result++;
+      temp.splice(-4, 4);
+    }
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(ingredient) {
  const temp = [];
  const hamburger = "1231";
  let result = 0;

  for (let i = 0; i < ingredient.length; i++) {
    temp.push(ingredient[i]);

    const checkLastFouringredient = temp.slice(-4).join("");

    if (checkLastFouringredient === hamburger) {
      result++;
      temp.splice(-4);
    }
  }

  return result;
}
```
- 풀이 잘 봤습니다 🙇‍♂️ temp(stack)을 이용한 풀이 잘봤습니다! 그런데 변수 선언할 때 항상 const (상수)를 상단으로 빼주시고 그다음 let(가변) 변수를 작성해주세요 ! 
- splice(-4, 4)아니여도 그냥 splice(-4)로 하셔도 됩니다!
- 아래는 제 풀이인데 reduce와 햄버거인지 체크하는 함수를 분리해서 풀어봤습니다.
```js
function solution(ingredient) {
  const stack = [];

  return ingredient.reduce((acc, cur) => {
    stack.push(cur);

    if (isHamburger(stack)) {
      stack.splice(-4);
      return acc + 1;
    }
    return acc;
  }, 0);
}

function isHamburger(stack) {
  const stackLength = stack.length;
  const hamburger = stack.slice(stackLength - 4, stackLength).join("");
    
  return hamburger === '1231';
}
```

Reply:
- 헉 변수위치 주의하겠습니다!!
- 아 저는 풀면서도 스택이라는걸 인지하지 못했는데 생각해보니 그렇네요..! 아직 연결시키는게 어렵습니다🥲
- reduce로 자주 푸시는데 활용법이 정말 다양하네요..! 참고 해서 저도 좀 더 응용해봐야겠습니다 감사합니다👍👍🙇‍♂️